consensus/core/src/synchronizer.rs

```diff
@@ -700,6 +717,18 @@ impl<C: NetworkClient, V: BlockVerifier, D: CoreThreadDispatcher> Synchronizer<C
         Ok(())
     }
 
+    fn is_commit_lagging(&self) -> (bool, CommitIndex, CommitIndex) {
+        let last_commit_index = self.dag_state.read().last_commit_index();
+        let quorum_commit_index = self.commit_vote_monitor.quorum_commit_index();
+        let commit_threshold = last_commit_index
+            + self.context.parameters.commit_sync_batch_size * COMMIT_LAG_MULTIPLIER;
+        (
+            commit_threshold < quorum_commit_index,
```

Comment:
nit: it might be better to inline this function at callsite or extract more logic into the function, since basically all local variables are returned.